### PR TITLE
Fix formatting of speeds in networkctl

### LINKED ISSUE #33076: `networkctl status` truncates link speeds
### systemd version the issue has been seen with

255

### Used distribution

NixOS

### Linux kernel version used

6.8.9

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

I got a 2.5G switch, and used `networkctl status` to check the link speed. I would have expected it to show 2.5Gbps.

### Unexpected behaviour you saw

It shows `Speed: 2Gbps`, rather than `Speed: 2.5Gbps`. `ethtool` properly reports `Speed: 2500Mb/s`.

### Steps to reproduce the problem

Connect a 2.5Gbps  network interface to a 2.5Gbps switch.

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
$ networkctl status 3
● 3: enp89s0
                   Link File: /nix/store/9cxd17xnmw0bi8n4nf722ysqj2bjlh8s-systemd-255.4/lib/systemd/network/99-default.link
                Network File: /etc/systemd/network/99-ethernet-default-dhcp.network
                       State: routable (configured)
                Online state: online                                                                                       
                        Type: ether
                        Path: pci-0000:59:00.0
                      Driver: igc
                      Vendor: Intel Corporation
                       Model: Ethernet Controller I226-LM
            Hardware Address: 58:47:ca:xx:cc:xx (Shenzhen Meigao Electronic Equipment Co.,Ltd)
                         MTU: 1500 (min: 68, max: 9216)
                       QDisc: mq
IPv6 Address Generation Mode: eui64
    Number of Queues (Tx/Rx): 4/4
            Auto negotiation: yes
                       Speed: 2Gbps
                      Duplex: full
                        Port: tp
                     Address: 172.20.42.106 (DHCP4 via 172.20.42.1)
                              2001:7d0:xxxx:xxxx::54c
                              fe80::5a47:caff:fexx:xxxx
                     Gateway: 172.20.42.1
                              fe80::ca7f:xxxx:xxxx:xxxx (ASUSTek COMPUTER INC.)
                         DNS: 172.20.42.1
                              fd42:172:2042::1
           Activation Policy: up
         Required For Online: yes
             DHCP4 Client ID: IAID:0x6404957c/DUID
           DHCP6 Client IAID: 0x6404957c
           DHCP6 Client DUID: DUID-EN/Vendor:0000ab114e255af0172a72da
                Connected To: n/a on port g6
                              n/a on port 58:47:ca:77:10:de

Mai 29 17:27:15 ws systemd-networkd[1770]: enp89s0: Gained IPv6LL
Mai 29 17:27:16 ws systemd-networkd[1770]: enp89s0: DHCPv6 address 2001:7d0:xxxx:xxxx::54c/128 (valid for 23min 11s, preferred for 23min 11s)
Mai 29 17:27:43 ws systemd-networkd[1770]: enp89s0: Lost carrier
Mai 29 17:27:43 ws systemd-networkd[1770]: enp89s0: DHCP lease lost
Mai 29 17:27:43 ws systemd-networkd[1770]: enp89s0: DHCPv6 lease lost
Mai 29 17:27:47 ws systemd-networkd[1770]: enp89s0: Gained carrier
Mai 29 17:27:47 ws systemd-networkd[1770]: enp89s0: DHCPv4 address 172.20.42.106/25, gateway 172.20.42.1 acquired from 172.20.42.1
Mai 29 17:27:47 ws systemd-networkd[1770]: enp89s0: DHCPv6 address 2001:7d0:xxxx:xxxx::54c/128 (valid for 22min 40s, preferred for 22min 40s)
```

```
$ ethtool enp89s0                  
Settings for enp89s0:
	Supported ports: [ TP ]
	Supported link modes:   10baseT/Half 10baseT/Full
	                       100baseT/Half 100baseT/Full
	                       1000baseT/Full
	                       2500baseT/Full
	Supported pause frame use: Symmetric
	Supports auto-negotiation: Yes
	Supported FEC modes: Not reported
	Advertised link modes:  10baseT/Half 10baseT/Full
	                       100baseT/Half 100baseT/Full
	                       1000baseT/Full
	                       2500baseT/Full
	Advertised pause frame use: Symmetric
	Advertised auto-negotiation: Yes
	Advertised FEC modes: Not reported
	Speed: 2500Mb/s
	Duplex: Full
	Auto-negotiation: on
	Port: Twisted Pair
	PHYAD: 0
	Transceiver: internal
	MDI-X: off (auto)
	Supports Wake-on: pumbg
	Wake-on: g
        Current message level: 0x00000007 (7)
                               drv probe link
	Link detected: yes
```

### PR DESCRIPTION
**Pull request by keszybz**
Fixes #33076.

### CODE REVIEW

File: src/basic/format-util.c [hunk 1]
```diff
@@ -70,15 +70,17 @@ char *format_bytes_full(char *buf, size_t l, uint64_t t, FormatBytesFlag flag) {
 
         for (size_t i = 0; i < n; i++)
                 if (t >= table[i].factor) {
-                        if (flag & FORMAT_BYTES_BELOW_POINT) {
+                        uint64_t remainder = i != n - 1 ?
+                                (t / table[i + 1].factor * 10 / table[n - 1].factor) % 10 :
```
Comment (yuwata) [inline src/basic/format-util.c:74]: I think
```
t / table[i + 1].factor / 100 % 10
```
is simpler.

File: src/basic/format-util.c [hunk 2]
```diff
@@ -70,15 +70,17 @@ char *format_bytes_full(char *buf, size_t l, uint64_t t, FormatBytesFlag flag) {
...
+                                (t * 10 / table[i].factor) % 10;
```
Comment (yuwata) [inline src/basic/format-util.c:75]: `()` is redundant.

File: src/shared/format-table.c
Comment (yuwata) [inline src/shared/format-table.c:1]: commit message of the third commit: 2.5gps -> 2.5Gbps

**Review by yuwata — APPROVED** [review]

File: src/basic/format-util.c (hunk 1 above)
Comment (keszybz) [reply src/basic/format-util.c:74]: We have two tables. So the dividend is 1000 or 1024, i.e. we'd need `/ 100` or `/ 102` here (ignoring rounding).

File: src/basic/format-util.c (hunk 2 above)
Comment (keszybz) [reply src/basic/format-util.c:75]: Strictly speaking, yes. But I think the formula is much easier to read with the additional parens.

File: src/shared/format-table.c
Comment (keszybz) [reply src/shared/format-table.c:1]: Fixed.

**Comment by bluca** [comment]
Is this needed for 256?

**Comment by yuwata** [comment]
This is a bug fix.

**Review by bluca — APPROVED** [review]

**Pull request merged**